src/DistributedWriteAheadLog/Record.cpp

```diff
@@ -44,8 +59,17 @@ RecordPtr Record::read(const char * data, size_t size)
     /// FIXME, more graceful version handling
     assert(Record::version(flags) == VERSION);
 
-    DB::NativeBlockInputStream input{rb, 0};
-
-    return std::make_shared<Record>(Record::opcode(flags), input.read());
+    if (unlikely(Record::compression(flags)))
+    {
+        DB::CompressedReadBuffer compressed_in = DB::CompressedReadBuffer(rb);
```

Comment:
I still didn't get an answer for my previous question : what is the default compression algorithm here ? can we explicitly choose the compression algorithm here ?

Reply:
CompressedReadBuffer read the codec from the header of the input ReadBuffer, therefore no need to specify codec explicitly. Hence the codec is determined when write data, the default codec it used is "LZ4" which is specified explicitly when create CompressedWriteBuffer

Reply:
If the codec is already self-contained in the binary data. We don't need reserve 2 bits for the  compression codec ?